apiserver/facades/agent/upgrader/upgrader.go

```diff
@@ -212,3 +200,99 @@ func (u *UpgraderAPI) DesiredVersion(ctx context.Context, args params.Entities)
 	}
 	return params.VersionResults{Results: results}, nil
 }
+
+// SetTools is responsible for updating a a set of entities reported agent
+// version.
+func (u *UpgraderAPI) SetTools(
```

Comment:
Same comments regarding naming and direct use of `AuthOwner` instead of the `canReadWrite` created in register.